### PR TITLE
(0013647) Add support for learned_cidrs_approval for aws_tgw_vpn_conn

### PR DESCRIPTION
**Pull request by zhiqiangjin**

### CODE REVIEW

File: aviatrix/resource_aviatrix_aws_tgw_vpn_conn.go [hunk 1]
```diff
@@ -203,10 +220,52 @@ func resourceAviatrixAwsTgwVpnConnRead(d *schema.ResourceData, meta interface{})
...
+			err := client.DisableVpnConnectionLearnedCidrsApproval(awsTgwVpnConn)
+			if err != nil {
+				return fmt.Errorf("failed to disable learned cidrs approval: %s", err)
+			}
+		}
+		d.SetPartial("enable_learned_cidrs_approval")
+	}
+
+	d.SetId(awsTgwVpnConn.TgwName + "~" + awsTgwVpnConn.VpnID)
+	return resourceAviatrixAwsTgwVpnConnRead(d, meta)
```
Comment (arvindsree) [inline aviatrix/resource_aviatrix_aws_tgw_vpn_conn.go:266]: use the defered read to handle failures

**Review by arvindsree — APPROVED** [review]

File: aviatrix/resource_aviatrix_aws_tgw_vpn_conn.go (hunk 1 above)
Comment (zhiqiangjin) [reply aviatrix/resource_aviatrix_aws_tgw_vpn_conn.go:266]: Only one thing, which is to create an aws_tgw_vpn_conn, is done in create function. Deferred read is not needed for this case.

Comment (arvindsree) [reply aviatrix/resource_aviatrix_aws_tgw_vpn_conn.go:266]: What if the disable actions fails but gives a warning and the state is actually false?

You can avoid the additional terraform refresh

Comment (arvindsree) [reply aviatrix/resource_aviatrix_aws_tgw_vpn_conn.go:266]: It's a good practice to follow everywhere

**Pull request merged**